botocore/regions.py

```diff
@@ -510,19 +514,43 @@ def _validate_account_id_endpoint_mode(self):
             )
             raise InvalidConfigError(error_msg=error_msg)
 
+    def resolve_credential_scope_builtin(
+        self, builtin_configured, builtin_value
+    ):
+        """Resolve the ``AWS::Auth::CredentialScope`` builtin."""
+        if not builtin_configured or self._credentials is None:
+            return None
+
+        if builtin_value is not None:
+            return builtin_value
+
+        frozen_creds = self._credentials.get_frozen_credentials()
+        return frozen_creds.scope
+
 
 class EndpointBuiltinResolver:
     """Resolves endpoint builtins during endpoint construction."""
 
-    def __init__(self, resolver_map):
+    def __init__(self, resolver_map, uses_builtin_data_path=True):
         self._resolver_map = resolver_map
+        self._uses_builtin_data_path = uses_builtin_data_path
```

Comment:
We could consider renaming this to prevent confusion between and endpoint builtin parameter and builtin data. Maybe we change the name to `uses_custom_data` and flip the bool around.